AndroidSdk/AvdManager/AvdManager.cs

```diff
@@ -59,45 +60,64 @@ public override FileInfo FindToolPath(DirectoryInfo androidSdkHome)
 
 		internal override string SdkPackageId => "emulator";
 
-		public void Create(string name, string sdkId, string device = null, string path = null, bool force = false, string sdCardPath = null, string sdCardSize = null)
+		public void Create(string name, string sdkId, string device = null, string path = null, bool force = false, string sdCardPath = null, string sdCardSize = null) =>
+			Create(name, sdkId, new AvdCreateOptions { Device = device, Path = path, Force = force, SdCardPathOrSize = string.IsNullOrEmpty(sdCardPath) ? sdCardSize : sdCardPath });
+
+		public void Create(string name, string sdkId, AvdCreateOptions options)
 		{
+			if (options == null)
+				options = new AvdCreateOptions();
+
 			var args = new List<string> {
 				"create", "avd", "-n", name, "-k", $"\"{sdkId}\""
 			};
 
-			if (!string.IsNullOrEmpty(device))
+			if (!string.IsNullOrEmpty(options.Device))
 			{
 				args.Add("--device");
-				args.Add($"\"{device}\"");
+				args.Add($"\"{options.Device}\"");
 			}
 
-			if (!string.IsNullOrEmpty(sdCardPath))
+			if (!string.IsNullOrEmpty(options.SdCardPathOrSize))
 			{
 				args.Add("-c");
-				args.Add($"\"{sdCardPath}\"");
+				args.Add($"\"{options.SdCardPathOrSize}\"");
 			}
 
-			if (!string.IsNullOrEmpty(sdCardSize))
+			if (!string.IsNullOrEmpty(options.Abi))
 			{
-				args.Add("-c");
-				args.Add($"\"{sdCardSize}\"");
+				args.Add("-b");
+				args.Add($"\"{options.Abi}\"");
 			}
 
-			if (force)
+			if (!string.IsNullOrEmpty(options.Skin))
+			{
+				args.Add("--skin");
+				args.Add($"\"{options.Skin}\"");
+			}
+
+			if (options.Force)
 				args.Add("--force");
 
-			if (!string.IsNullOrEmpty(path))
+			if (!string.IsNullOrEmpty(options.Path))
 			{
 				args.Add("-p");
-				args.Add($"\"{path}\"");
+				args.Add($"\"{options.Path}\"");
 			}
 
-			run(args.ToArray());
+			run(args.ToArray(), input =>
+			{
+				// Continuously send "ENTER" to respond to the prompt:
+				// "Do you wish to create a custom hardware profile? [no]"
+				// Sending "no" too soon will cause an error...
+				input.WriteLine();
+				input.Flush();
+			});
```

Comment:
This happens in some situations when you are on a new machine and there is no "hardware profile"... Not sure what that means, but it usually is on CI.